### ISSUE TITLE
Support more kart animations (especially to allow transitions between different states)

### ISSUE THREAD

**Issue by Alayan-stk-2** [supertuxkart/stk-code#4942]
Request by Zaqraven.

Here are some of the currently missing transitions : 
- [ ] From reverse-left / reverse-right to kart moving forward steering left / right
- [ ] From kart steering left / right to reverse-left / reverse-right (there is a transition only from driving straight into reversing)
- [ ] Transition from jump animation to normal animation

**Comment by ZAQraven** [comment]
About jump animation, to be clarified, when the jump animation ended while still in the air, it would not be looped until the kart landed and played jump-out.
Also, if the kart landed before the jump animation ended, immediately transited into a jump-out animation.

**Comment by Erikcht** [comment]
One of the things that leaves Tux kart is unpleasant is the animation of the characters.
Mario Kart 64 did not need to use 3D models for animation and used a technique called "Sprites".

https://github.com/supertuxkart/stk-code/assets/8018995/0845b43d-f496-43be-a01b-84b59e0589ce


The same technique of Sprites used in Doom 1 & 2, and all the children of Doom as Duke Nukem and Recent Boomer Shooters.


The problem of the 3D model of the tux kart characters is that the dolls are not yet animated very well.
Their animation behaves very hard in their cars, it's strange.

https://github.com/supertuxkart/stk-code/assets/8018995/ae9088c3-e70d-405b-b5c9-d83962e7afb2

**Comment by Alayan-stk-2** [comment]
Yes, it's true that many (most ?) karts have subpar animations even for the animations that are already supported by the game.

**Comment by ZAQraven** [comment]
Regarding the animation of winning/losing after finishing the race, I am thinking of removing:
- winning-to-straight
- losing-to-straight
- removing their looping

Meaning, we only need (for the finish line's animation):
- winning-start
- winning-end
- neutral-start
- neutral-end
- losing-start
- losing-end

Don't want the character doing that monotonously, just with this will be enough.

**Comment by Gustanator** [comment]
Maybe we can play the end animation after some 5 secs (or some time else) of the race finish, as is very strange (at least for me) the character continues the finish race animation after 30 secs (or more :P)

**Comment by Alayan-stk-2** [comment]
> Regarding the animation of winning/losing after finishing the race, I am thinking of removing:
> 
>     * winning-to-straight
> 
>     * losing-to-straight
> 
>     * removing their looping

All these have a purpose.

The "to straight" animations were initially removed when we looked at the new animations as we didn't understand their point, but they are used in soccer after a goal, removing them created a regression.

Although I think the case could be made that the new happy animation could be used for positive character reactions after soccer goals. Not so sure about the hit animation as a sad animation after own goals, though. The win/lose animations are often too involved/long, and having them continue to play for 3 seconds after the game has resumed is questionable.

For the winning, neutral and losing animations, having a start, loop start and end is crucial to allow animations to have an introductory sequence. This one is non-negotiable, it's already widely used and it is crucial for animation quality.
